kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/identity/FileStoredIdentityManager.java

```diff
@@ -0,0 +1,118 @@
+
+    protected static final Logger log = Logger.getLogger(FileStoredIdentityManager.class);
+
+    private static final String IDENTITY_FILE = System.getProperty(ROUTER_IDENTITY_FILE, "users.properties");
+    private File userFile;
+
+    public FileStoredIdentityManager() {
+        userFile = new File(IDENTITY_FILE);
+        log.infof("Using <%1$s> as user repository", userFile.getAbsolutePath());
+
+    }
+
+    @Override
+    public Account verify(Account account) {
+        return account;
+    }
+  
+    @Override
+    public Account verify(Credential credential) {
+        return null;
+    }
+  
+    @Override
+    public Account verify(String id, Credential credential) {
+        if (!(credential instanceof PasswordCredential)) {
+            return null;
+        }
+        PasswordCredential passwordCredential = (PasswordCredential) credential;
+        KieServerInstanceAccount account = getAccount(id);
+        String challenge = HashUtil.hash(id, new String(passwordCredential.getPassword()));
+        if (account != null && challenge.equals(account.getHash())) {
+            log.infof("Authentication succesfully for %1$s", id);
+            return account;
+        }
+        log.infof("Authentication failed for %1$s", id);
+        return null;
+    }
+
+    private KieServerInstanceAccount getAccount(String id) {
+        Map<String, KieServerInstanceAccount> users = new HashMap<>();
+        Properties file = loadKieServerInstanceData();
+        file.stringPropertyNames().forEach(name -> users.put(name, new KieServerInstanceAccount(name, file.getProperty(name))));
+        return users.get(id);
+    }
+
+    @Override
+    public void addKieServerInstance(String kieServerInstanceId, String password) {
+        Properties data = loadKieServerInstanceData();
+        data.put(kieServerInstanceId, HashUtil.hash(kieServerInstanceId, password));
+        saveKieServerInstanceData(data);
+    }
+
+    @Override
+    public void removeKieServerInstance(String kieServerInstanceId) {
+        Properties data = loadKieServerInstanceData();
+        data.remove(kieServerInstanceId);
+        saveKieServerInstanceData(data);
+    }
+
+    private Properties loadKieServerInstanceData() {
+        Properties properties = new Properties();
+        try (InputStream is = new FileInputStream(userFile)) {
+            properties.load(is);
+        } catch (IOException e) {
+            log.errorf(e, "An error ocurred. File <%1$s> is a directory, it does not exist or it cannot be opened", userFile);
+        }
+        return properties;
+    }
+
+    private void saveKieServerInstanceData(Properties data) {
+        try (OutputStream out = new FileOutputStream(userFile)) {
+            data.store(out, "Kie Server Router");
+        } catch (IOException e) {
+            log.errorf(e, "An error ocurred. File <%1$s> is a directory, it does not exist or it cannot be opened", userFile);
```

Comment:
As @sutaakar pointed out, when opening a FileOutputStream the possible reasons for FileNotFoundException are:
```
Throws:
FileNotFoundException - if the file exists but is a directory rather than a regular file, does not exist but cannot be created, or cannot be opened for any other reason
```

So the line should be

`"An error ocurred. File <%1$s> is a directory, it cannot be created or it cannot be opened"`

Reply:
fine